fastapi_sso/sso/base.py

```diff
@@ -345,6 +345,9 @@ async def process_login(
             current_url = str(url)
 
         current_path = f"{url.scheme}://{url.netloc}{url.path}"
+        if self.provider == "linkedin":
+            # linkedin needs 'client_secret' parameter
+            params["client_secret"] = self.client_secret
```

Comment:
Oh no, they really do require users to put client secret in the query params? That's sad... 😞 
I'd prefer some other way of addressing this issue rather then adding `if self.provider == ...` for every exception. What about some overridable method `update_query_params`?

Reply:
Hi @tomasvotava,
yes you're right, my method is not the best one or cleaner one.
In the next days I'll try to find a better way to accomplish that.
At now I've 2 proposals:
    - expect the the user adds explicit 'params' dictionary with the client_id when call 'login_process()'
    - as you suggest, any provider furnish 'update_query_params() -> dict' in which returns a dictionary
For the second idea I think something like this

inside base.py
`
def update_query_params(self, param : dict) -> dict:
     return params
   
async def process_login(
        self,
        code: str,
        request: Request,
        *,
        params: Optional[Dict[str, Any]] = None,
        additional_headers: Optional[Dict[str, Any]] = None,
        redirect_uri: Optional[str] = None,
    ) -> Optional[OpenID]:
    
    ....    
    params = self._update_query_params(params or {})
    ...
`


linkedin.py
`
def update_query_params(self, param : dict) -> dict:
     return params.update({'client_id': self.client_id})
`

Reply:
Hi @tomasvotava,

Please check my changes, I hope they go well.